services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -1033,20 +1034,38 @@ public void deleteStore(
       sendAdminMessageAndWaitForConsumed(clusterName, storeName, message);
 
       // Deleting ACL needs to be the last step in store deletion process.
-      if (store != null) {
-        if (!store.isMigrating()) {
-          cleanUpAclsForStore(storeName, VeniceSystemStoreType.getEnabledSystemStoreTypes(store));
-        } else {
-          LOGGER.info("Store: {} is migrating! Skipping acl deletion!", storeName);
-        }
-      } else {
-        LOGGER.warn("Store object for {} is missing! Skipping acl deletion!", storeName);
-      }
+      deleteAclsForStore(store, storeName);
+    } catch (AdminMessageConsumptionTimeoutException timeoutException) {
+      LOGGER.info(
+          "Timed out while waiting for delete store admin message to be consumed for store: {} in cluster: {}",
+          storeName,
+          clusterName,
+          timeoutException);
+      deleteAclsForStore(store, storeName);
+    } catch (Exception e) {
+      LOGGER.info("Caught an exception when deleting store {} in cluster {}", storeName, clusterName, e);
     } finally {
       releaseAdminMessageLock(clusterName, storeName);
     }
   }
 
+  /**
+   * Deletes the acls associated with a store
+   * @param store
+   * @param storeName
+   */
+  protected void deleteAclsForStore(Store store, String storeName) {
+    if (store != null) {
+      if (!store.isMigrating()) {
+        cleanUpAclsForStore(store.getName(), VeniceSystemStoreType.getEnabledSystemStoreTypes(store));
+      } else {
+        LOGGER.info("Store: {} is migrating! Skipping acl deletion!", storeName);
+      }
+    } else {
+      LOGGER.warn("Store object for {} is missing! Skipping acl deletion!", storeName);
+    }
```

Comment:
```suggestion
    if (store == null) {
      LOGGER.warn("Store object for {} is missing! Skipping acl deletion!", storeName);
      return;
    }

    if (!store.isMigrating()) {
      cleanUpAclsForStore(store.getName(), VeniceSystemStoreType.getEnabledSystemStoreTypes(store));
    } else {
      LOGGER.info("Store: {} is migrating! Skipping acl deletion!", storeName);
    }
```
super nit: I know you're just refactoring lines around, but how about this instead? It would save one level of indentation.